### DISCUSSION TITLE
Accept a callback for `isAuthenticated`

### DISCUSSION THREAD

**Discussion by brandonpittman** [sergiodxa/remix-auth#216] [Ideas]
This is connected to the process of refreshing tokens, but even with remix-auth, I have to have a separate `requireUser` function to refresh tokens. What about adding a callback option to `isAuthenticated` that could run whenever the function is called and returns anything other than `null`?

**Comment by sergiodxa** [comment]
Could you provide an example API and use case?

**Reply by brandonpittman** [reply]
## API

An API might be:

```ts
let onAuthenticated = (request: Request, data: User) => { /* refresh API token, etc. */}

auth.isAuthenticated(request, {successRedirect: "", onAuthenticated})
```

I'd like to provide it once when setting up the authenticator, but this seems like the only reasonable way to get the request and the `sessionKey` data into the callback.

## Use Case

As for the use case, you may find it to be out of scope, but I would like to wrap up token refreshing per [the method you outlined](https://sergiodxa.com/articles/working-with-refresh-tokens-in-remix). As it stands, because I need to refresh tokens, I can't do this:

```ts
return await authenticator.isAuthenticated(request, {
    successRedirect: "/dashboard",
  });
```

I have to do the whole token refresh dance in a separate custom `authenticate` function. If I could package it up with the standard remix-auth instance, it would be convenient.

**Reply by chinanderm** [reply]
This would be extremely handy to have.
